lib/_tls_wrap.js

```diff
@@ -228,7 +228,7 @@ function initRead(tls, wrapped) {
  */
 
 function TLSSocket(socket, options) {
-  this._tlsOptions = options;
+  this._tlsOptions = options || { isServer: false };
```

Comment:
I'll make `isServer: false` by default (set to `false` if `options` is `undefined` or `options.isServer` is `undefined`).
Regarding the context - [TLS API](https://nodejs.org/api/tls.html#tls_class_tls_tlssocket) says the rest of the options (besides `isServer`) are optional and seems like `_wrapHandle()` creates a `secureContext` if such is not provided.